packages/commerce-sdk-react/src/auth/index.ts

```diff
@@ -282,8 +283,52 @@ class Auth {
         DATA_MAP[name].callback?.(storage)
     }
 
+    private delete(name: AuthDataKeys) {
+        const {key, storageType} = DATA_MAP[name]
+        const storage = this.stores[storageType]
+        storage.delete(key)
+    }
+
     getDnt() {
-        return this.get(this.dntCookieName)
+        const dntCookieVal = this.get(DNT_COOKIE_NAME)
+        let dntStatus = undefined
+        if (dntCookieVal !== '1' && dntCookieVal !== '0') {
+            this.delete(DNT_COOKIE_NAME)
+        } else {
+            dntStatus = Boolean(Number(dntCookieVal))
+        }
+        return dntStatus
+    }
+
+    async setDnt(preference: boolean | null, secondsUntilExpire?: number) {
+        let dntCookieVal = String(Number(preference))
+        // Use defaultDNT if defined. If not, use SLAS default DNT
+        if (preference === null) {
+            dntCookieVal = this.defaultDnt ? String(Number(this.defaultDnt)) : '0'
+        }
+        // Set the cookie once to include dnt in the access token and then again to set the expiry time
+        this.set(DNT_COOKIE_NAME, dntCookieVal, {
+            ...getDefaultCookieAttributes(),
+            secure: true
+        })
+        const accessToken = this.getAccessToken()
+        if (accessToken !== '') {
+            const {dnt} = this.parseSlasJWT(accessToken)
+            if (dnt !== dntCookieVal) {
+                await this.refreshAccessToken()
+            }
+        } else {
+            await this.refreshAccessToken()
+        }
+        if (preference !== null) {
+            this.set(DNT_COOKIE_NAME, dntCookieVal, {
+                ...getDefaultCookieAttributes(),
+                secure: true,
+                ...(secondsUntilExpire !== undefined && {
+                    expires: Number(this.get('refresh_token_expires_in')) / 86400
```

Comment:
Nit: Let's put a constant for 86400 to give it a context. e.g const SECONDS_IN_DAY = 24 * 60 * 60